### DISCUSSION TITLE
Markdown stream

### DISCUSSION THREAD

**Discussion by AndiLeni** [Textualize/rich#3287] [Q&A] [answered]
Hello,

I am trying to print a stream from an ai api directly to the console and render it as markdown.

The code I tries looks like this:

```python
raw_text = ""

try:
    for chunk in client.chat_stream(
        model="mistral-medium",
        messages=[
            ChatMessage(
                role="system",
                content=f"Answer the user's questions based on the below context. If the context doesn't contain any relevant information to the question, don't make something up and just say 'I don't know': <context>${combined_string}</context>",
            ),
            ChatMessage(role="user", content=query),
        ],
    ):
        if chunk.choices[0].delta.content is not None:
            # print(chunk.choices[0].delta.content, end="")
            raw_text += chunk.choices[0].delta.content
            md = Markdown(raw_text)
            console.print(md, end="\r")
except Exception as e:
    print(f"Error: {e}")
```

However, this produces a new line after each received chunk:

![image](https://github.com/Textualize/rich/assets/13385771/c5e10de3-505b-4946-9f71-86067d390b9a)

Is there any way to not create a new line and rather overwrite the current text?

Thanks and kind regards

**Comment by willmcgugan** [comment, marked as answer]
You can use [Live](https://rich.readthedocs.io/en/latest/live.html), but for something like this you may be better off using [Textual](https://textual.textualize.io/)

**Reply by AndiLeni** [reply]
Thanks for your quick answer. I do not have much experience with textualize, so please bear with my lack of knowledge. 
Could you point me the direction (docs, example) for this issue? I am still figuring out how using textualize may solve this problem for this specific usecase.

**Comment by willmcgugan** [comment]
This project does something similar. https://github.com/darrenburns/elia

Suggest you join the Discord server if you have any questions.

**Reply by AndiLeni** [reply]
This looks like a good repo to take inspiration from. Thanks :)
